File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/registry/resolver/ChatInfoResolver.java
package ru.tggc.telegrambotcore.registry.resolver;

import org.springframework.stereotype.Component;
import ru.tggc.telegrambotcore.annotation.params.ChatInfo;
import ru.tggc.telegrambotcore.dto.ChatDto;

import java.lang.reflect.Parameter;

@Component
public class ChatInfoResolver implements ParameterResolver<ChatDto> {

    @Override
    public boolean supports(Parameter p) {
        return p.isAnnotationPresent(ChatInfo.class) && p.getType().equals(ChatDto.class);
    }

    @Override
    public ChatDto resolve(Parameter parameter, HandlerCtx ctx) {
        return new ChatDto(ctx.chat().id(), ctx.chat().title());
    }
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/registry/resolver/ChatInfoResolver.kt
package ru.tggc.telegrambotcore.registry.resolver

import org.springframework.stereotype.Component
import ru.tggc.telegrambotcore.annotation.params.ChatInfo
import ru.tggc.telegrambotcore.dto.ChatDto
import java.lang.reflect.Parameter

@Component
class ChatInfoResolver : ParameterResolver<ChatDto?> {
    override fun supports(p: Parameter?): Boolean? =
        p?.isAnnotationPresent(ChatInfo::class.java) == true && p.getType() == ChatDto::class.java

    override fun resolve(parameter: Parameter?, ctx: HandlerCtx): ChatDto =
        ChatDto(ctx.chat!!.id(), ctx.chat.title())

}
